core/server/data/migrations/versions/3.29/05-add-member-constraints.js

```diff
@@ -0,0 +1,156 @@
+            // first drop the key
+            await knex.schema.alterTable('members_labels', (table) => {
+                table.dropForeign('member_id');
+                table.dropIndex('member_id', 'members_labels_member_id_foreign');
+            });
+            // then re-add with ON DELETE CASCADE
+            await knex.schema.alterTable('members_labels', (table) => {
+                table.foreign('member_id').references('members.id').onDelete('CASCADE');
+            });
+        }
+
+        const labelIdConstraint = dbConstraints.find(constraint => constraint.CONSTRAINT_NAME === 'members_labels_label_id_foreign');
+        if (labelIdConstraint && labelIdConstraint.DELETE_RULE === 'CASCADE') {
+            logging.warn('Skipping ON DELETE CASCADE for "members_labels_label_id_foreign" constraint - already set');
+        } else if (labelIdConstraint) {
+            logging.info('Adding ON DELETE CASCADE to "members_labels_label_id_foreign" constraint');
+            // first drop the key
+            await knex.schema.alterTable('members_labels', (table) => {
+                table.dropForeign('label_id');
+                table.dropIndex('label_id', 'members_labels_label_id_foreign');
+            });
+            // then re-add with ON DELETE CASCADE
+            await knex.schema.alterTable('members_labels', (table) => {
+                table.foreign('label_id').references('labels.id').onDelete('CASCADE');
+            });
+        }
+
+        // stripe tables have not had any indexes/constraints in the past, add them now with ON DELETE CASCADE
+
+        const [membersStripeCustomersIndexes] = await knex.raw('SHOW INDEXES FROM members_stripe_customers');
+        const [membersStripeCustomersSubscriptionsIndexes] = await knex.raw('SHOW INDEXES from members_stripe_customers_subscriptions');
+
+        if (membersStripeCustomersIndexes.find(index => index.Key_name === 'members_stripe_customers_member_id_foreign')) {
+            logging.warn('Skipping "members_stripe_customers_member_id_foreign" foreign key constraint creation - already exists');
+        } else {
+            logging.info('Adding "members_stripe_customers_member_id_foreign" foreign key constraint');
+            await knex.schema.alterTable('members_stripe_customers', (table) => {
+                table.foreign('member_id').references('members.id').onDelete('CASCADE');
+            });
+        }
+
+        if (membersStripeCustomersIndexes.find(index => index.Key_name === 'members_stripe_customers_customer_id_unique')) {
+            logging.warn('Skipping "members_stripe_customers_customer_id_unique" index creation - already exists');
+        } else {
+            logging.info('Adding "members_stripe_customers_customer_id_unique" index');
+            await knex.schema.alterTable('members_stripe_customers', (table) => {
+                table.unique('customer_id');
```

Comment:
I think we're missing add the `subscription_id` unique constraint in this migration too!

Reply:
👌 I'll get that added